### DISCUSSION TITLE
Enable event notification to kafka using TLS

### DISCUSSION THREAD

**Discussion by udhayd** [minio/minio#15803] [Q&A]
Hi team , 

I have a requirement to enable event notification to kafka for minio buckets and need to configure through TLS.

I made a attempt to configure a notify_kafka with kafka server details by uploading certificate and key file on "tls_client_cert" and "tls_client_key" parameters through 'mc' client command from pod and tls files are placed inside a pod to configure the event notification. 

However getting error   "no such file  found".

Could you please guide me to configure the same , I don't see these details in the  official documentation as well

**Comment by vadmeste** [comment]
@udhayd did you provide the absolute path of the tls certificates in the pod when configuring with mc ?

**Reply by udhayd** [reply]
Hi , 

As mentioned , Cert & key are stored in POD but still its not able to recognize it . Please correct if i am wrong in configuring it .

[root@fileupload-59b8494c9c-kzq8l opt]# ls -lt /opt/svc-ssrv-uat-datp.key /opt/svc-ssrv-uat-datp.crt
-rwxr-xr-x. 1 root root 3421 Oct  5 06:56 /opt/svc-ssrv-uat-datp.key
-rwxr-xr-x. 1 root root 9054 Oct  5 06:54 /opt/svc-ssrv-uat-datp.crt
[root@fileupload-59b8494c9c-kzq8l opt]#
[root@fileupload-59b8494c9c-kzq8l opt]# mc admin config set minio notify_kafka:primary \
>    brokers="kafka-uat-ssl-broker:9094" \
>    topic="minio-events" \
>    tls_skip_verify="on" \
>    sasl="off" \
>    sasl_password="" \
>    sasl_username="" \
>    tls="on" \
>    client_tls_cert="/opt/svc-ssrv-uat-datp.crt" \
>    client_tls_key="/opt/svc-ssrv-uat-datp.key" \
>    version="" \
>    queue_dir="" \
>    queue_limit="0" \
>    comment=""
mc: <ERROR> Unable to set 'notify_kafka:primary brokers=kafka-uat-ssl-broker:9094 topic=minio-events tls_skip_verify=on sasl=off sasl
_password= sasl_username= tls=on client_tls_cert=/opt/svc-ssrv-uat-datp.crt client_tls_key=/opt/svc-ssrv-uat-datp.key version= queue_dir= queue_l
imit=0 comment=' to server: open /opt/svc-ssrv-uat-datp.crt: no such file or directory.
[root@fileupload-59b8494c9c-kzq8l opt]#

**Reply by harshavardhana** [reply]
which pod? @udhayd are you running minio on the same pod where you have the certs stored?

**Reply by udhayd** [reply]
Pod Configured with MC Client binary to configure the Minio bucket Notification .

**Reply by harshavardhana** [reply]
where is `mc `and where is `minio` and where are the certs stored? - in terms of the k8s pods.

**Reply by udhayd** [reply]
minio service is deployed in "minio" namespace & i am trying to execute mc command from pod in "test" namespace. Cert & key is stored in pod in "test" namespace.

**Reply by vadmeste** [reply]
> minio service is deployed in "minio" namespace & i am trying to execute mc command from pod in "test" namespace. Cert & key is stored in pod in "test" namespace.

The certs need to be found in _all_ MinIO pods

**Reply by harshavardhana** [reply]
> minio service is deployed in "minio" namespace & i am trying to execute mc command from pod in "test" namespace. Cert & key is stored in pod in "test" namespace.

certs must be present on MinIO namespace and on its pods for it to work, mc cannot send your certs from local test namespace to minio namespace.

**Reply by udhayd** [reply]
Hi , 

Could you please let me know where should we place the certs in minio namespace should we configure the Certs in Configmap (or) place the certs & key in /export filesystem mounted .

Minio Storage is mounted as "/export" filesystem in minio pod.

pod/minio-0   1/1     Running   0          24d
pod/minio-1   1/1     Running   0          24d
pod/minio-2   1/1     Running   0          24d
pod/minio-3   1/1     Running   0          24d

NAME                    TYPE        CLUSTER-IP       EXTERNAL-IP   PORT(S)    AGE
service/minio           ClusterIP   172.26.48.27     <none>        9000/TCP   280d
service/minio-console   ClusterIP   172.26.109.192   <none>        9001/TCP   280d
service/minio-svc       ClusterIP   None             <none>        9000/TCP   280d

NAME                     READY   AGE
statefulset.apps/minio   4/4     24d

**Reply by udhayd** [reply]
Hello Team , 

Could you please help on this , Thanks !
